packages/grid/x-data-grid/src/components/cell/GridCell.tsx

```diff
@@ -90,12 +123,127 @@ const useUtilityClasses = (ownerState: OwnerState) => {
 
 let warnedOnce = false;
 
+// GridCellWrapper is a compatibility layer for the V6 cell slot. If we can use the more efficient
+// `GridCellV7`, we should. That component is a merge of `GridCellWrapper` and `GridCell`.
+// TODO(v7): Remove the wrapper & cellV6 and use the cellV7 exclusively
```

Comment:
TODO: This probably broke the visualization demo.

Reply:
I'm fine with duplicating it for now (considering it a backward compatibility cost).
Is the difference substantial when using 2 components?

Reply:
IIRC, on the test-case 1, it was adding 0.5ms per scroll event on my brand new machine, so the cost could go to 1-1.5ms on average & low-end devices. If we target 60 FPS, we have 16ms total budget per scroll event (because scroll triggers right after a repaint according to the spec), if we assume the browser does its thing in 6ms we have 10ms for us.

So using 2 components for each cell can chew 5-15% of our budget, because cells are the most numerous components on screen.

NOTE: Actually it could be a bit higher than that. The additional 0.5ms I observed was for the `React.createElement` function, but having more nodes in the tree also affects the diffing cost.

Reply:
Thanks for the details 👍